### ISSUE TITLE
Connection failure during image building for the apt-get step with binderhub

### ISSUE THREAD

**Issue by nikl11** [jupyterhub/binderhub#1316]
**TL;DR:** Image build process in my binderhub fails on commands accessing internet, like apt-get install, wget etc. I have the same issues with a basic docker build, and solve it by adding --network host. Is there a way to add --network host to the binderhub building parameters?

**### Bug description**
Hello, I built and started a new binderhub exactly as the "Zero to Binderhub" manual says, and despite never ever dealing with helm previously, I go it running at http://my-public-IP:31356. BUT, if I copy@paste a link to a git repository with jupyter notebook that I would like to build into an image and run on my binderhub, everything seems to be building fine until docker commands that require an internet connection are reached. For example step 1-3 during the building phase go fine 

Building conda environment...
Step 1/51 : FROM buildpack-deps:bionic
...
Step 2/51 ENV DEBIAN_FRONTEND=noninteractive
....
Step 3/51 RUN apt-get -qq update && apt-get -qq install....

At this  step the build just freezes, during any command that requires to dowload something from the internet, like apt-get, wget, git clone etc.

**#### Expected behaviour**
Finish the whole build and run the jupyter notebook from the image locally.

**#### Actual behaviour**
I think this is the same error when trying to build docker images with docker build on the same machine where this binderhub run (I should maybe add that it is a virtual machine running in our company's OpenStack cloud, so the whole networking structure is probably more complicated and right now beyond my abilities), during docker build I also freeze on command like apt-get install with Connection failure. BUT I can fix it by adding --network host as a docker build parameter. After that building works flawlesly. But how do I add --network host into binderhub, or is there any other workaround how to achieve what --network host does? The problem is that each binderhub build creates its own pod where it builds, so if I could make any changes to a specific pod, I dont have how to target it because it doesnt exist until it is already building and failing.

**### How to reproduce**
If you ever encountered that docker build fails on commands like apt-get with Connection failure, but adding --network host to docker build parameters help, then you probably have the same issue when trying to build images in binderhub. Otherwise I am not sure how to reproduce.

I currently cannot try to run binderhub on any other machine, I have tried it on my desktop PC with kubeadm kubernetes, and I have problems with the "tiller pod" not starting properly, so the whole helm doesnt work properly because of that. Then tried to run binderhub on my laptop under the minikube kubernetes, which seems to run fine, but because minikube creates some sort of a virtualbox instance as a container/pod to run everything in, I dont know which IP and port to enter into my browser to access the binderhub, I have tried every combination of IP I see in ifconfig output and every port that kubectl service returns, but I was not able to access the cluster, probably some sort of a bridge or proxy is needed to setup between minikube and host?  

**## Your personal set up**
For binderhub I used "zero to binderhub" guide, I can access it in my browser. But the server I installed it on is a virtual OpenStack cloud instance, running Ubuntu 18.04 as its operating system. I am NOT a network guru and I have been combatting network issues ever since I started this job creating and preparing cloud instances for our customers. So definitely I have to somehow reddirect or allow the image building process an access to the outside internet world, like what --network host does, but how the heck do I do it for binderhub (that unfortunately uses helm to describe its properties and settings, which is another super complicated and super annoying level of experties next to docker, kubernetes, cloud instances and its networking etc.

**Comment by nikl11** [comment]
OK the problem has been found, our server used different MTU limit (packet size limit) than what docker had defaulty set, so once I added the proper mtu value into /etc/docker/daemon.json 

{
     "mtu": 1442
}

and restarted docker by sudo service docker restart, the building process now has proper access to the internet.
